### ISSUE TITLE
Kodein 7 typeToken UnsupportedOperationException for byte[]

### ISSUE THREAD

**Issue by eygraber** [Kodein-Framework/Kodein-DI#307]
I am in the process of updating Kodein to v7, and I ran into a crash with the `generic` function. I was previously using the erased artifact everywhere (aside from specific cases where I needed generic support).

The crash is occurring here:

```
bind<GCMParameterSpec>() with factory { iv: ByteArray ->
    GCMParameterSpec(GCM_TAG_LENGTH, iv)
}
```

[Source for GCMParameterSpec](https://android.googlesource.com/platform/libcore/+/refs/heads/master/ojluni/src/main/java/javax/crypto/spec/GCMParameterSpec.java)

Here's the stacktrace:

```
Caused by: java.lang.UnsupportedOperationException: Unsupported type libcore.reflect.GenericArrayTypeImpl: byte[]
        at org.kodein.type.TypeTokensJVMKt.typeToken(typeTokensJVM.kt:85)
        at co.myapp.crypto.Crypto_moduleKt$cryptoModule$1.invoke(crypto_module.kt:92)
        at co.myapp.crypto.Crypto_moduleKt$cryptoModule$1.invoke(Unknown Source:2)
        at org.kodein.di.internal.DIBuilderImpl.import(DIBuilderImpl.kt:55)
        at org.kodein.di.DI$Builder$DefaultImpls.import$default(DI.kt:301)
        at co.myapp.MyApp$di$1.invoke(MyApp.kt:111)
        at co.myapp.MyApp$di$1.invoke(MyApp.kt:93)
        at org.kodein.di.internal.DIImpl$Companion.newBuilder(DIImpl.kt:22)
        at org.kodein.di.internal.DIImpl$Companion.access$newBuilder(DIImpl.kt:21)
        at org.kodein.di.internal.DIImpl.<init>(DIImpl.kt:19)
        at org.kodein.di.DI$Companion$lazy$1.invoke(DI.kt:443)
        at org.kodein.di.DI$Companion$lazy$1.invoke(DI.kt:425)
        at kotlin.SynchronizedLazyImpl.getValue(LazyJVM.kt:74)
        at org.kodein.di.LazyDI.getBaseDI(Unknown Source:2)
        at org.kodein.di.LazyDI.getContainer(lateinit.kt:48)
        at org.kodein.di.DIAwareKt.getDirect(DIAware.kt:236)
        at co.myapp.AppInitProvider.onCreate(AppInitProvider.kt:46)
        at android.content.ContentProvider.attachInfo(ContentProvider.java:2092)
        at android.content.ContentProvider.attachInfo(ContentProvider.java:2066)
        at android.app.ActivityThread.installProvider(ActivityThread.java:6983)
        at android.app.ActivityThread.installContentProviders(ActivityThread.java:6528) 
        at android.app.ActivityThread.handleBindApplication(ActivityThread.java:6445) 
        at android.app.ActivityThread.access$1300(ActivityThread.java:219) 
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1859) 
        at android.os.Handler.dispatchMessage(Handler.java:107) 
        at android.os.Looper.loop(Looper.java:214) 
        at android.app.ActivityThread.main(ActivityThread.java:7356) 
        at java.lang.reflect.Method.invoke(Native Method) 
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:492) 
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:930) 
```

**Comment by eygraber** [comment]
I tried changing the factory parameter to `Array<Byte>` but that didn't work either.

I had to use:

`internal inline class ByteArrayHolder(val value: ByteArray)`

:expressionless:

**Comment by romainbsl** [comment]
Sorry for the inconvenience... 
This is linked to our new Type system, I'll look into it ASAP.

**Comment by eygraber** [comment]
Any update on this or https://github.com/Kodein-Framework/Kodein-DI/issues/308

**Comment by romainbsl** [comment]
Still investigating, This is on our todo list for the next week.

**Comment by romainbsl** [comment]
fixed in [7.2](https://github.com/Kodein-Framework/Kodein-DI/releases/tag/7.2.0) with Kodein-Type 1.4
